src/audio/port_eu.c

```diff
@@ -304,88 +300,54 @@ extern void *D_8019403C;
 void func_800CBAB4(void) {
     D_800EA3A0 = 0;
     D_800EA3A4 = 0;
-    osCreateMesgQueue(D_800EA3A8, &D_80194020, 1);
-    osCreateMesgQueue(D_800EA3AC, &D_80194028, 4);
-    osCreateMesgQueue(D_800EA3B0, &D_80194038, 1);
-    osCreateMesgQueue(D_800EA3B4, &D_8019403C, 1);
+    osCreateMesgQueue(D_800EA3A8[0], &D_80194020, 1);
+    osCreateMesgQueue(D_800EA3A8[1], &D_80194028, 4);
+    osCreateMesgQueue(D_800EA3A8[2], &D_80194038, 1);
+    osCreateMesgQueue(D_800EA3A8[3], &D_8019403C, 1);
 }
-#else
-GLOBAL_ASM("asm/non_matchings/audio/port_eu/func_800CBAB4.s")
-#endif
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-extern u8 D_800EA3A0;
-extern ? D_80193820;
-
-void func_800CBB48(s32 arg0, s32 *arg1) {
-    void *temp_v0;
+extern struct EuAudioCmd D_80193820[0x100];
 
-    temp_v0 = (D_800EA3A0 * 8) + &D_80193820;
-    temp_v0->unk0 = arg0;
-    temp_v0->unk4 = *arg1;
-    D_800EA3A0 = D_800EA3A0 + 1;
+void func_800CBB48(s32 arg0, s32 *arg1){
+  struct EuAudioCmd *cmd = &D_80193820[D_800EA3A0 & 0xff];
+  cmd->u.first = arg0;
+  cmd->u2.as_u32 = *arg1;
+  D_800EA3A0++;
 }
-#else
-GLOBAL_ASM("asm/non_matchings/audio/port_eu/func_800CBB48.s")
-#endif
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-void func_800CBB48(? *); // extern
-
-void func_800CBB88(? arg1) {
-    func_800CBB48(&arg1);
+void func_800CBB88(u32 arg0, f32 arg1) {
+    func_800CBB48(arg0, (s32*) &arg1);
```

Comment:
If it works it works, but it seems really strange to typecast a float pointer to an int pointer

Reply:
Yes please do `func_800CBB48(arg0, &arg1);` as that matches without the cast to s32

Reply:
Actually, this is from sm64. It's fine as is.

Reply:
Interesting

Reply:
I think the reasoning for this cast is for the programmer. So when you read it, it's like "Hey, this f32 gets cast to s32" rather than it being there for programmatic purposes.